app/controllers/auth/AuthenticatedController.scala

```diff
@@ -90,12 +89,12 @@ trait AuthenticatedController extends FrontendBaseController with AuthorisedFunc
   }
 
   def ctAuthorisedCredID(body: => String => Future[Result])(implicit request: Request[AnyContent]): Future[Result] = {
-    baseFunction.retrieve(credentials)(cr => body(cr.providerId)) recover authErrorHandling()
+    baseFunction.retrieve(credentials)(cr => body(cr.get.providerId)) recover authErrorHandling()
```

Comment:
This will throw a `None.get` Exception if there is no credential retrieved. It would be better to patten match one the Option and then handle the `Some(cr)` and `None` separately, so that you can log out something useful and handle gracefully. Similar to how the other methods handle this.